deploy/scripts/install.sh

```diff
@@ -43,6 +43,8 @@ if [[ -z "$USER_OBJ_ID" ]]
     echo "To get USER_OBJ_ID, run 'az ad signed-in-user show --query id --output tsv'"
 fi
 
+az feature register --namespace "Microsoft.ContainerService" --name "EnableWorkloadIdentityPreview"
```

Comment:
Is Workload Identity available beyond the commercial cloud yet? E.g., sovereign clouds? Pod Identity is an option if not--it's definitely deprecated, though, and supposed to be replaced with workload identity so I wouldn't change/add support unless we're missing workload identity support.

Reply:
Good catch. I will try to get an answer to this before merging this PR.

Reply:
Worked with @karthick-rn today to validate if workload identity is available in Fairfax. We went through creating a new AKS Cluster and configuring the Managed Identity and Service Account. Then federated the Svc Account with the Managed Identity. Looks like the workload identity is available in Fairfax. 

One finding from the exercise was that the Kubernetes version 1.25.4 is not available. The latest available version is 1.25.2. I will change the version in this PR.

Reply:
Good news about workload identity (and federated credentials for managed identities) in sovereign clouds.

Chasing k8s versions can be tough since they release frequently and also deprecate frequently. Setting the k8s version to an empty string in the templates should cause it to use the default version defined for the particular cloud and region. I think that might be the way to go--if you want to choose a specific version, then you can do so by setting the k8s version explicitly in your template.

Reply:
As per our discussion, I have set the Minimum AKS version to 1.22.0 (Workload Identity supported version). Added logic in the install.sh to get the latest version and error out if the version is less than 1.22.0.